scalding-macros/src/main/scala/com/twitter/scalding/macros/impl/ordser/CaseClassOrderedBuf.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ Copyright 2014 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.twitter.scalding.macros.impl.ordser
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+import com.twitter.scalding._
+import com.twitter.scalding.serialization.OrderedSerialization
+import com.twitter.bijection.macros.impl.IsCaseClassImpl
+
+object CaseClassOrderedBuf {
+  def dispatch(c: Context)(buildDispatcher: => PartialFunction[c.Type, TreeOrderedBuf[c.type]]): PartialFunction[c.Type, TreeOrderedBuf[c.type]] = {
+    case tpe if tpe.typeSymbol.isClass && tpe.typeSymbol.asClass.isCaseClass && !tpe.typeConstructor.takesTypeArgs =>
+      CaseClassOrderedBuf(c)(buildDispatcher, tpe)
+  }
+
+  def apply(c: Context)(buildDispatcher: => PartialFunction[c.Type, TreeOrderedBuf[c.type]], outerType: c.Type): TreeOrderedBuf[c.type] = {
+    import c.universe._
+    def freshT(id: String) = newTermName(c.fresh(id))
+
+    val dispatcher = buildDispatcher
+    val elementData: List[(c.universe.Type, TermName, TreeOrderedBuf[c.type])] =
+      outerType
+        .declarations
+        .collect { case m: MethodSymbol if m.isCaseAccessor => m }
+        .map { accessorMethod =>
+          val fieldType = accessorMethod.returnType
+          val b: TreeOrderedBuf[c.type] = dispatcher(fieldType)
+          (fieldType, accessorMethod.name.toTermName, b)
+        }.toList
+
+    new TreeOrderedBuf[c.type] {
+      override val ctx: c.type = c
+      override val tpe = outerType
+      override def compareBinary(inputStreamA: ctx.TermName, inputStreamB: ctx.TermName) =
+        ProductLike.compareBinary(c)(inputStreamA, inputStreamB)(elementData)
+
+      override def hash(element: ctx.TermName): ctx.Tree = q"$element.hashCode"
```

Comment:
It does not look like you are recursively calling the hash function here. So, if an inner type has a `def hash()` that is not just `.hashCode`, it would be lost here, right?